### PR TITLE
#158559792 Send emails to users

### PR DESCRIPTION
**Pull request by chadwalt**
#### What does this PR do?
This PR adds the functionality of sending emails to users who have shown interest in given categories.

#### Description of the task to be completed?
Currently, when a user subscribes to an interest an email is not sent to them. This PR adds a functionality of sending emails to users via the email they signup with. An email will be sent out every 24 hours containing a page that a user can read and learn something new.

#### How should this be manually tested?
Before, sending emails one needs to configure the mailer by providing the following environment variables `EMAIL_DOMAIN`, `EMAIL`, `EMAIL_USERNAME`,and `EMAIL_PASSWORD`

This can be tested by executing this command `rake send_emails:mail`

#### What are the relevant pivotal tracker stories?
[#158559792](https://www.pivotaltracker.com/story/show/158559792)
#### Screenshots
N/A

### CODE REVIEW

**Comment by arnawldo** [comment]
Hey @chadwalt , I noticed there are no instructions related to setting up email credentials. Can you make the testing instructions more elaborate for someone to test locally. Also I think you can add instructions for testing the cron job locally, preferably with a shorter interval

File: lib/mailer.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+users.each do |value|
+  if value.interests
+    body = "<p>Here are things to learn about </p>" \
+      "<p>Title: #{pages[value.interests[0]][0].title}</p>" \
+      "<p>Description: #{pages[value.interests[0]][0].description}</p>"
+
+    MailHelpers.send_mail(value.email, 'Things to learn', body)
+  end
+end
```
Comment (arnawldo) [inline lib/mailer.rb]: Why is the functionality not placed inside a function? Plus where are the tests?

Comment (chadwalt) [reply lib/mailer.rb]: I have not placed that functionality inside a function because I will not be reusing that code.

**Comment by chadwalt** [comment]
@arnawldo I have updated the readme with how to set up the email configurations.

File: lib/mailer.rb (hunk 1 above)
Comment (arnawldo) [reply lib/mailer.rb]: Its bad practice to have dangling code in a file. It is close to impossible to write tests for this for example, and its not scalable. Placing the logic in a module or class with functions should always be priority

**Comment by arnawldo** [comment]
@chadwalt please double check README. I can't see any changes

File: lib/mailer.rb (hunk 1 above)
Comment (arnawldo) [reply lib/mailer.rb]: @chadwalt

Comment (chadwalt) [reply lib/mailer.rb]: I have put all the logic in a class

**Comment by arnawldo** [comment]
LGTM

**Pull request merged**